internal/pkg/utils/network_linux.go

```diff
@@ -187,20 +187,35 @@ func DoNetworking(
 				return fmt.Errorf("failed to set net.ipv6.conf.lo.disable_ipv6=0: %s", err)
 			}
 
-			// No need to add a dummy next hop route as the host veth device will already have an IPv6
-			// link local address that can be used as a next hop.
-			// Just fetch the address of the host end of the veth and use it as the next hop.
-			addresses, err := netlink.AddrList(hostVeth, netlink.FAMILY_V6)
-			if err != nil {
-				logger.Errorf("Error listing IPv6 addresses for the host side of the veth pair: %s", err)
-				return err
+			// Retry several times as the LL can take a several micro/miliseconds to initialize and we may be too fast
+			// after these sysctls
+			var err error
+			var addresses []netlink.Addr
+			for i := 0; i < 10; i++ {
+				// No need to add a dummy next hop route as the host veth device will already have an IPv6
+				// link local address that can be used as a next hop.
+				// Just fetch the address of the host end of the veth and use it as the next hop.
+				addresses, err = netlink.AddrList(hostVeth, netlink.FAMILY_V6)
+				if err != nil {
+					logger.Errorf("Error listing IPv6 addresses for the host side of the veth pair: %s", err)
+				}
+
+				if len(addresses) < 1 {
+					// If the hostVeth doesn't have an IPv6 address then this host probably doesn't
+					// support IPv6. Since a IPv6 address has been allocated that can't be used,
+					// return an error.
+					err = fmt.Errorf("failed to get IPv6 addresses for host side of the veth pair")
+				}
+				if err == nil {
```

Comment:
what err is checking this condition?

Reply:
Whoops misread.
This is the one checking if we can break after less than 10 retries.

Reply:
It terminates the retry loop if there is no error at both 198 and 207

Reply:
Yeah, check my 2nd reply, I was looking at the wrong err.

It just makes it not retry 10 times (as in that case there may have just been time.Sleep(500ms)) but continue as fast as possible.

Reply:
Should I still change it to flip the `if len(addresses)` or is this ok? :) (missed that due to the edit)

Reply:
I think it's OK, it's how the old code worked so no _need_ to change it.